agent/consul/auto_encrypt.go

```diff
@@ -0,0 +1,126 @@
+package consul
+
+import (
+	"fmt"
+	"net"
+	"net/rpc"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/consul/agent/connect"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/consul/tlsutil"
+	"github.com/hashicorp/go-msgpack/codec"
+)
+
+func (c *Client) AutoEncrypt(servers []string, port int, token string) (*structs.SignResponse, string, error) {
+	errFn := func(err error) (*structs.SignResponse, string, error) {
+		return nil, "", err
+	}
+
+	if len(servers) == 0 {
+		return errFn(fmt.Errorf("No servers to request AutoEncrypt.Sign"))
+	}
+
+	// We don't provide the correct host here, because we don't know any
+	// better at this point. Apart from the domain, we would need the
+	// ClusterID, which we don't have. This is why we go with the domain
+	// the first time. Subsequent CSRs will have the correct Host.
+	id := &connect.SpiffeIDAgent{
+		Host:  strings.TrimSuffix(c.config.Domain, "."),
```

Comment:
put something obviously static in here, instead of trying to guess the trust domain.